### ISSUE TITLE
[2025/04/09] Machine Learning 輪講

### ISSUE THREAD

**Issue by chimuichimu** [wantedly/machine-learning-round-table#292]
## Why

Machine Learning 輪講は最新の技術や論文を追うことで、エンジニアが「技術で解決できること」のレベルをあげていくことを目的にした会です。

prev. #290

## What

話したいことがある人はここにコメントしましょう！
面白いものを見つけた時点でとりあえず話すという宣言だけでもしましょう！

**Comment by ghibney** [comment]
## [360Brew: A Decoder-only Foundation Model for Personalized Ranking and Recommendation](https://arxiv.org/abs/2501.16450)

### Journal / Conference
\- (arXiv)

### 概要
* reranker に LLM を利用して、特徴量エンジニアリング不要で高精度かつ汎用的な推薦システムを提案している、linkedin による技術レポート
   * 8つ以上のサーフェス（推薦表示箇所）における、30以上のタスクでモデルをオフライン評価し、production かそれ以上の精度を実現
   * インタラクション履歴の少ないコールドスタートユーザーに対しては、本番モデルよりも高い性能
   * 時間経過による性能低下が本番モデルよりも小さい

### 背景
* linkedin の推薦システムは多数のモデルからなる複雑なアーキテクチャになっており、維持・改善に多大な労力を要している（特に特徴量エンジニアリング）
* IDベースの特徴量に依存するため、新規ユーザーやアイテムに対するコールドスタート問題が発生しやすい
* 特定のドメインやタスクに特化しており、新しいサーフェイス・タスクへの汎化が困難
* モデル間の複雑な依存関係（DAG）の管理が必要

### 目的
* LLM を用いて、LinkedIn の既存レコメンデーションシステムが抱える複雑さや開発・維持コストの問題を解決するアプローチとその有効性を示すこと

### Method
* レコメンデーションタスクを、下記のように many shot の ICLとして定義している
<img width="686" alt="Image" src="https://github.com/user-attachments/assets/3721ceaa-9d8d-493f-9961-bd73299c31cc" />

* Mixtral 8x22 pre-trained MoE に、LinkedInの一次データ（メンバープロファイル、求人情報、投稿、インタラクション履歴など）を用いて、追加学習・ファインチューニングしている

### Result

* データスケールしている
<img width="771" alt="Image" src="https://github.com/user-attachments/assets/9b46f724-a83f-4c76-9a95-2c2ca1921dad" />

* 学習に用いられていないドメインのタスクについても、Production の専用モデルと比較して同等以上の精度が出ている
<img width="872" alt="Image" src="https://github.com/user-attachments/assets/f9713dd2-941b-4b66-ba24-a153d4a3861f" />

* production と比べて、インタラクションが少ない場合の性能が特に高い
<img width="511" alt="Image" src="https://github.com/user-attachments/assets/613e4a99-56e3-4b62-8fc6-d2c51921ab69" />

* 時間経過によるデータ分布のシフトが起きても、精度に対する影響が production と比較して小さい
<img width="760" alt="Image" src="https://github.com/user-attachments/assets/c5a87ef6-6114-42e2-9c79-4f6b7c920636" />

**Comment by Hayashi-Yudai** [comment]
## [Auto Encoding Neural Process for Multi-interest Recommendation](https://micxyb.github.io/papers/NPRec.pdf)

- ドメイン：シーケンシャル推薦
- 着目した課題："Individual preferences possess the diverse property and dynamic tendency"
    - ユーザーの嗜好が複数ある場合にも、適切な推薦ができるようにしたい
    - ユーザーの興味は時間とともに変化していく場合にも、適切な推薦ができるようにしたい
- 先行研究：
    - SIR：ユーザーの嗜好を全部ひとまとめにして1つのベクトルで表現する
        - 嗜好を区別したり、時間変化を表現するのは難しい
    - MIR：嗜好ごとにベクトルを作る
        - 作るベクトルの数は人が設定する必要があるらしい。そのため、適切なパフォーマンスを得るのが難しい

### 提案手法

予測を決定論的に行うのではなく、(分布を予測するタスクにすり替えることで)信頼度を含めて確率的に予測するようにする

<img width="616" alt="Image" src="https://github.com/user-attachments/assets/97022019-0ce9-4fdd-ba2e-5521ec0d6ee6" />

モデルのアーキテクチャは↓

<img width="1130" alt="Image" src="https://github.com/user-attachments/assets/dbe61d31-f40b-45af-8639-d2025686ccff" />

- 青枠で示されている "Latent Encoder" が確率的な予測を担当する部分
- 1段目でSequentialモデルを学習、2段目でLatent EncoderやDeterministic Encoderを使ってアイテムとラベルの関係の埋込を作成
- 最後にClassificationタスクをといて、出力スコアの降順に並べることでTop-N推薦にする

### 結果

- 従来のSIR, MIRモデルと比較して大きな性能改善を実現
- 1段目のSequentialモデルで使っているMambaを除くと精度がガタ落ち
    - 2段目のEncoderだけではシーケンス情報をうまく学習できない
- Sequentialモデルの部分を他のSIRモデルに置き換えると、SIR単体と比べて大きな精度改善
    - 2段目のモデルは、これまでのシーケンシャル推薦モデル一般に適用可能で汎用性が高い

**Comment by chimuichimu** [comment]
## [Bridging Search and Recommendation in Generative Retrieval Does One Task Help the Other?](https://dl.acm.org/doi/10.1145/3640457.3688123)

### Summary
検索と推薦という2つのタスクを1つの生成モデルで統一的に扱うアプローチを提案する、Spotifyの論文
### Journal / Conference
Recsys 2024
### Background
従来、検索エンジンとレコメンデーションシステムは異なる信号（検索ではクエリ、レコメンドでは履歴）を基に動作しており、別々のモデルや特徴量を用いて構築されてきた。しかし、多くのオンラインプラットフォームではこれらのシステムが共存しており、情報を共有・統合することで性能向上が期待される
### Motivation

検索と推薦は異なるユーザー行動を捉えている。これらの補完的な関係性を活用し、より良いアイテムの表現を学習したい

具体的には以下2つの仮説を検証する

- 統一的な学習により、アイテムの popularity を正則化できるのでは
- 検索の情報を活用することで推薦の精度を向上することができるのでは（逆もまた然り）

### Method
検索のクエリ-アイテム対と、レコメンドのユーザー-アイテム対の両方を用いて、単一の生成検索モデルをマルチタスク学習で訓練。入力から直接アイテムIDを生成する手法を使っているらしい。

> We use Huggingface’s implementation of T5 [43] (google/flan-t5-base) and train all generative models for 5 epochs with a learning rate

とあるのでモデルはT5を使っている？
以下は別のページから持ってきたもの

![Image](https://github.com/user-attachments/assets/44e0e81a-c663-4b2c-980c-3a01eaddfde9)

### Evaluation

シミュレーションデータと実データを用いて仮説を検証

- 実データ
	- ML: MovieLens 25M
	- MPD: Spotify の PlayList データセット
	- PodCasts: Spotify の Podcast 視聴ログのデータセット

### Result

※ 実データの結果のみを記載

- 多くのシナリオで個別タスク特化モデルを上回る精度が得られた
    - Recall@30が平均16%改善
- （popularity の正則化についても触れられているが、ちゃんと読めていない）
